l10n_fi_invoice/views/account_invoice_templates.xml

```diff
@@ -553,6 +548,7 @@
             report_type="qweb-pdf"
             name="l10n_fi_invoice.report_invoice_finnish_translate"
             file="l10n_fi_invoice.report_invoice_finnish_translate"
+            print_report_name="('-'.join(((object.refund_invoice_id.number or object.number or ''),object.company_id.display_name,object.partner_id.display_name)).replace(' ','-').replace(',','').strip('-'))"
```

Comment:
What happens here if you drop `object.refund_invoice_id.number`? Shouldn't `object.number` return the invoice number of the active invoice whether it's the original invoice or the refund?

Reply:
If `object.refund_invoice_id.number` is dropped then draft credit notes don't get a number in their file name. Their `number` field is filled in when they get validated.
Validated credit notes will get the value of their own `number` field in the filename, which is different from the originating invoice.

Reply:
But if `object.refund_invoice_id.number is not False` and `object.number is not False` then `object.refund_invoice_id.number` is the primary selection even for validated credit notes. Right?

Reply:
Yes

Reply:
I kind of feel like this should be something like 
`print_report_name="('-'.join(((object.number or 'draft'),object.company_id.display_name,object.partner_id.display_name)).replace(' ','-').replace(',','').strip('-'))"
`
So if the `invoice.number` is not set, it will be called just draft.

Reply:
Then the validated credit note file name won't contain the originating invoice number, though. Wasn't that one of the requirements?

Reply:
But then the credit note will have an identical name than the original invoice. I can't see that in the requirements.

When I think about it, the best would be `type-(refund-)invoice-number-company-customer.pdf`, so:
Invoices: `Invoice-12345-Avoin-Systems-Client-Oy`
Credit Notes: `Refund-12346-12345-Avoin-Systems-Client-Oy.pdf`

Or did @svantesuominen have something else on mind?

Reply:
I have no particular feelings about the format of the name, but the logic is getting long and complicated. Can you extract the logic into a separate method so we don't need to have a hard to read string thing?
```
print_report_name="object._get_invoice_pdf_filename()"
```